Source/Extensions/Blazorise.SignaturePad/wwwroot/signaturepad.js

```diff
@@ -176,4 +185,42 @@ function getImageDataURL(sigpad, options) {
     }
 
     return sigpad.toDataURL("image/png", options.imageQuality || 1);
+}
+
+window.onresize = resizeAllCanvas;
+
+function resizeAllCanvas() {
+    if (_instances) {
+        for (let i = 0; i < _instances.length; i++) {
+            const instance = _instances[i];
+
+            if (instance) {
+                resizeCanvas(instance.sigpad, instance.element);
+            }
+        }
+    }
+}
+
+function resizeCanvas(sigpad, canvas) {
+    if (!sigpad) {
+        return;
+    }
+
+    if (!canvas) {
+        return;
+    }
+
+    // When zoomed out to less than 100%, for some very strange reason,
+    // some browsers report devicePixelRatio as less than 1
+    // and only part of the canvas is cleared then.
+    const ratio = Math.max(window.devicePixelRatio || 1, 1);
+
+    const context = canvas.getContext("2d");
+
+    // This part causes the canvas to be cleared
+    canvas.width = canvas.offsetWidth * ratio;
+    canvas.height = canvas.offsetHeight * ratio;
+    context.scale(ratio, ratio);
+
+    sigpad.fromData(sigpad.toData());
```

Comment:
I guess this is a way to refresh? Do they not have a refresh or reset or whatever method? So you don't have to incur the costs of `toData` and `FromData`? Altought they might be negligible, I don't know.

Reply:
This is per their instructions so as far as I know there is nothing built in that does the same.